public/js/controllers/game.js

```diff
@@ -1,184 +1,267 @@
-            setTimeout(function(){
-              var link = document.URL;
-              var txt = 'Give the following link to your friends so they can join your game: ';
-              $('#lobby-how-to-play').text(txt);
-              $('#oh-el').css({'text-align': 'center', 'font-size':'22px', 'background': 'white', 'color': 'black'}).text(link);
-            }, 200);
-            $scope.modalShown = true;
-          }
+        $location.search({ game: game.gameID });
+        if (!$scope.modalShown) {
+          setTimeout(() => {
+            const link = document.URL;
+            const txt = 'Give the following link to your friends so they can join your game: ';
+            $('#lobby-how-to-play').text(txt);
+            $('#oh-el').css({ 'text-align': 'center', 'font-size': '22px', background: 'white', color: 'black' }).text(link);
+          }, 200);
+          $scope.modalShown = true;
         }
       }
-    });
+    }
+  });
+  $scope.countries =
+  [
+    {
+      country: 'General',
+      regionId: '58f4de8ef08434413b6aec50'
+    },
+    {
+      country: 'Nigeria',
+      regionId: '58ed5fbe75ebcefb68f19750'
+    },
+    {
+      country: 'USA',
+      regionId: '58f531a4f08434413b6aec51'
+    },
+    {
+      country: 'South Africa',
+      regionId: '58ed60a875ebcefb68f19751'
+    },
+    {
+      country: 'Kenya',
+      regionId: '58ed60a875ebcefb68f19752'
+    },
+    {
+      country: 'Uganda',
+      regionId: '58ed60a875ebcefb68f19753'
+    },
+    {
+      country: 'Ghana',
+      regionId: '58ed60a875ebcefb68f19754'
+    },
+    {
+      country: 'England',
+      regionId: '58ed620175ebcefb68f19769'
+    },
+    {
+      country: 'Spain',
+      regionId: '58ed620175ebcefb68f1976a'
+    },
+    {
+      country: 'Germany',
+      regionId: '58ed620175ebcefb68f1976b'
+    },
+    {
+      country: 'Sweden',
+      regionId: '58ed620175ebcefb68f1976c'
+    },
+    {
+      country: 'Denmark',
+      regionId: '58ed620175ebcefb68f1976d'
+    },
+    {
+      country: 'Italy',
+      regionId: '58ed620175ebcefb68f1976e'
+    },
+    {
+      country: 'France',
+      regionId: '58ed620175ebcefb68f1976f'
+    }
+  ];
 
-    if ($location.search().game && !(/^\d+$/).test($location.search().game)) {
-      console.log('joining custom game');
-      game.joinGame('joinGame',$location.search().game);
-    } else if ($location.search().custom) {
-      game.joinGame('joinGame',null,true);
+  $scope.selectedCountry = $scope.countries[0];
+
+  $scope.selectCountry = (region) => {
+    if (region) {
+      $http.post('/api/selected-region',
+           { regionId: '58f4de8ef08434413b6aec50' });
     } else {
-      game.joinGame();
+      const chosenCountry = angular.element(document
+            .querySelector('#selectedCountry')).val();
+      const chosen = chosenCountry.split(':')[1];
+      $scope.selectedCountryId = $scope.countries[chosen].regionId;
+      $http.post('/api/selected-region',
```

Comment:
can you move this $http post to userService, why don't we send the 
```
{
country: 'France',
regionId: '58ed620175ebcefb68f1976f'
 }
```
Then check if country is 'General' in the model, rather than hardcode the Id, what if the id changes, or someone deletes a char unknowingly, I would even prefer that the region data is stored on the server, and we make a GET request (just like Avatars)